lib/utils/documentdb_model.ts

```diff
@@ -153,4 +153,46 @@ export abstract class DocumentDbModel<
 
     return updatedDocument.mapRight(this.toRetrieved).mapRight(some);
   }
+
+  /**
+   * Upsert an attachment for a specified document.
+   * 
+   * @param documentId    the id of the document
+   * @param partitionKey  partition key of the document
+   * @param attachment    attachment object (contentType and media link)
+   */
+  public async attach(
+    documentId: string,
+    partitionKey: string,
+    attachment: DocumentDb.Attachment
+  ): Promise<Either<DocumentDb.QueryError, Option<TR>>> {
+    // we do not try to retrieve the document before attaching media,
+    // the operation will fail automatically on non existing documents
+    const attachmentMeta = await DocumentDbUtils.upsertAttachment(
+      this.dbClient,
+      DocumentDbUtils.getDocumentUri(this.collectionUri, documentId),
+      attachment,
+      { partitionKey }
+    );
+
+    if (attachmentMeta.isLeft) {
+      return left(attachmentMeta.left);
+    }
+
+    // return the document with attachment media information
+    return this.find(documentId, partitionKey);
+  }
+
+  public async getAttachments(
+    documentId: string,
+    options: DocumentDb.FeedOptions = {}
+  ): Promise<ReadonlyArray<DocumentDb.AttachmentMeta>> {
```

Comment:
I would return an Iterator here and let the client decide whether to fetch everything with `iteratorToArray` or lazily process the results.

Reply:
ok

Reply:
ok